frame/contracts/src/storage/meter.rs

```diff
@@ -441,21 +444,15 @@ where
 		// contract execution does conclude and hence would lead to a double charge.
 		self.total_deposit = deposit.clone();
 
+		// We need to make sure that the contract's account itself exists.
+		T::Currency::transfer(origin, contract, ed, Preservation::Preserve)?;
+
+		System::<T>::inc_providers(&contract);
```

Comment:
Need a comment here why this is a provider and not a `consumer` or `sufficient`. It was a consumer before.

Reply:
This has been moved to (and commented) within the `terminate` fn
https://github.com/paritytech/substrate/blob/8ed3b66d41dc8518351180c9420a0df52b34fd56/frame/contracts/src/exec.rs#L1278-L1284

Reply:
Okay by doing this we are able to send away all the balance and keep the account briefly alive. However, we should still add a `consumer` at account creation and remove it on termination as we did before (the provider here is on top). Reason being is that the system will be to brittle if we don't. Our guarantee is that: As long as a contract exists its account does exist. Without a consumer it hinges on us on calling transfer with the correct parameters. Something that we can't enforce if a contract calls into the runtime. With the consumer a correct runtime cannot remove the account.

Reply:
this has been resolved in https://github.com/paritytech/substrate/pull/14767